Assets/SequenceSDK/WaaS/Tests/EOAWalletToSequenceWalletTest.cs

```diff
@@ -0,0 +1,364 @@
+            adapter.OnDeployContractComplete += OnContractDeployed;
+            adapter.OnDeployContractFailed += OnContractNotDeployed;
+        }
+
+        int successfullSignatureCounter;
+        int failedSignatureCounter;
+
+        [Test]
+        public async Task TestSignMessageT()
+        {
+            successfullSignatureCounter = 0;
+            failedSignatureCounter = 0;
+            var signature = await adapter.SignMessage(chain, "TestSingature");
+            Assert.AreEqual(1, successfullSignatureCounter);
+            Assert.Zero(failedSignatureCounter);
+            Assert.IsNotNull(signature);
+            Assert.IsNotEmpty(signature);
+        }
+        void OnMessageSigned(string signature)
+        {
+            successfullSignatureCounter++;
+        }
+
+        void OnMessageFailed(string error)
+        {
+            failedSignatureCounter++;
+        }
+        [Test]
+        public async Task TestSendTransactionT()
+        {
+            var erc20 = await MakeERC20Transaction(await TestDeployERC20(_erc20ByteCode));
+
+            var erc721 = await MakeERC721Transaction(await TestDeployERC721(_erc721Bytecode));
+
+            var erc1155 = await MakeERC1155Transaction(await TestDeployERC1155(_erc1155ByteCode));
+
+            var delayedEncode = MakeDelayedEncodeApprove(contractAddress);
+
+
+            var txs = new Transaction[] { /*erc20,*/ erc721 , erc1155 , delayedEncode };
+            var txreturn = await adapter.SendTransaction(chain, txs, true);
+
+            if (txreturn is SuccessfulBatchTransactionReturn successfulBatchTransactionReturn)
+            {
+                foreach (var item in successfulBatchTransactionReturn.SuccessfulTransactionReturns)
+                {
+                    Assert.IsNotNull(item);
+                    Assert.IsNotNull(item.txHash);
+                    Assert.IsNotEmpty(item.txHash);
+                }
+            }
+            else if (txreturn is SuccessfulTransactionReturn successfulTransactionReturn)
+            {
+                Assert.IsNotEmpty(successfulTransactionReturn.txHash);
+            }
+            else if (txreturn is FailedTransactionReturn failedTransactionReturn)
+            {
+                Assert.Fail("Failed transaction return: "+ failedTransactionReturn.error );
+            }
+            else if (txreturn is FailedBatchTransactionReturn failedBatchTransactionReturn)
+            {
+                foreach (var item in failedBatchTransactionReturn.FailedTransactionReturns)
+                {
+                    Assert.Fail("Failed transaction return: " + item.error);
+                }
+                foreach (var item in failedBatchTransactionReturn.SuccessfulTransactionReturns)
+                {
+                    Assert.IsNotNull(item);
+                    Assert.IsNotNull(item.txHash);
+                    Assert.IsNotEmpty(item.txHash);
+                }
+            }
```

Comment:
This is actually unreachable since FailedBatchTransactionReturn is a type of FailedTransactionReturn (it should go above that check). Not a big deal on it's own but since we need to make a bit more changes anyways (resolve merge conflicts and comment erc20 back in) we might as well change this too